engine/merge_accessors.ml

```diff
@@ -17,7 +17,18 @@ and
   | AcTag of sym_value * int
 
 let rec merge : Target_sym_engine.constraint_tree -> constraint_tree =
-  let rec subst_pi (pi: Target_sym_engine.pi) : pi =
+  let rec map_sym_value (pi: Target_sym_engine.pi) : pi =
+    match pi.var with
+    | AcRoot v -> {var=AcRoot v; op=pi.op}
+    | AcField (s, i) ->
+      let inner = map_sym_value {var=s; op=pi.op} in
+      {var=AcField (inner.var, i); op=inner.op}
+    | AcTag (s, i) ->
+      let inner = map_sym_value {var=s; op=pi.op} in
+      {var=AcTag (inner.var, i); op=inner.op}
+    | AcAdd (_, _) -> assert false
```

Comment:
Maybe here an exception should be used rather than assert false

Reply:
There are two standard exceptions pre-declared in OCaml with helper functions to raise them: `failwith "foo"` to indicate error conditions during a program execution, and `invalid_arg "foo"` to indicate a programmer error in the choice of inputs (a precondition is violated). The latter would be appropriate here, but I move to merge the PR and move on.

(My work on intersection of constraints is unfinished but it does touch the constraint-tree structures as well, so having the current PR in master is good for me as well.)